tests/e2e/push_notifications/test_default_push_notification_support.py

```diff
@@ -0,0 +1,221 @@
+from notifications_app import create_notifications_app
+from utils import (
+    create_app_process,
+    find_free_port,
+    wait_for_server_ready,
+)
+
+
+@pytest.fixture(scope='session')
+def port_lock():
+    """Multiprocessing lock for acquiring available ephemeral ports."""
+    return Lock()
+
+
+@pytest.fixture(scope='module')
+def notifications_server(port_lock):
+    """
+    Starts a simple push notifications injesting server and yields its URL.
+    """
+    with port_lock:
+        host = '127.0.0.1'
+        port = find_free_port()
+        url = f'http://{host}:{port}'
+
+        process = create_app_process(create_notifications_app(), host, port)
+        process.start()
+        try:
+            wait_for_server_ready(f'{url}/health')
+        except TimeoutError as e:
+            process.terminate()
+            raise e
+
+    yield url
+
+    process.terminate()
+    process.join()
+
+
+@pytest_asyncio.fixture(scope='module')
+async def notifications_client():
+    """An async client fixture for calling the notifications server."""
+    async with httpx.AsyncClient() as client:
+        yield client
+
+
+@pytest.fixture(scope='module')
+def agent_server(
+    port_lock,
+    notifications_client: httpx.AsyncClient,
+):
+    """Starts a test agent server and yields its URL."""
+    with port_lock:
+        host = '127.0.0.1'
+        port = find_free_port()
+        url = f'http://{host}:{port}'
+
+        process = create_app_process(
+            create_agent_app(url, notifications_client), host, port
+        )
+        process.start()
+        try:
+            wait_for_server_ready(f'{url}/v1/card')
+        except TimeoutError as e:
+            process.terminate()
+            raise e
+
+    yield url
+
+    process.terminate()
+    process.join()
+
+
+@pytest_asyncio.fixture(scope='function')
+async def http_client():
+    """An async client fixture for test functions."""
+    async with httpx.AsyncClient() as client:
+        yield client
+
+
+@pytest.mark.asyncio
+async def test_notification_triggering_with_in_message_config_e2e(
+    notifications_server: str, agent_server: str, http_client: httpx.AsyncClient
+):
+    """
+    Tests push notification triggering for in-message push notification config.
+    """
+    # Send a message with a push notification config.
+    response = await http_client.post(
+        f'{agent_server}/v1/message:send',
+        json={
+            'configuration': {
+                'pushNotification': {
+                    'id': 'n-1',
+                    'url': f'{notifications_server}/notifications',
+                    'token': uuid.uuid4().hex,
+                },
+            },
+            'request': {
+                'messageId': 'r-1',
+                'role': 'ROLE_USER',
+                'content': [{'text': 'Hello Agent!'}],
+            },
+        },
+    )
+    assert response.status_code == 200
+    task_id = response.json()['task']['id']
```

Comment:
Consider parsing this into a response message rather than working with it as a dict.